File: app/src/main/java/com/example/myqueue/Repository.java
package com.example.myqueue;

import android.content.Context;

import androidx.lifecycle.LiveData;

import com.example.myqueue.DrawerActivity.ProgressBarCallBack;
import com.example.myqueue.api.CreateQueueModel;
import com.example.myqueue.api.GetMembersResponseBodyModel;
import com.example.myqueue.api.JoinQueueRequestBody;
import com.example.myqueue.api.SignUpRequestBody;
import com.example.myqueue.db.Queue;

import java.util.List;

public interface Repository {
    void signup(Context context, ProgressBarCallBack progressBarCallBack, SignUpRequestBody signUpRequestBody);

    void login(Context context, ProgressBarCallBack progressBarCallBack, String userName, String password);

    void logout(Context context);

    void getLoggedInUser(Context context);

    LiveData<List<Queue>> getMyQueues();

    void updateMyQueues(Context context, ProgressBarCallBack progressBarCallBack, int userId);

    LiveData<List<GetMembersResponseBodyModel>> getMembers();

    void updateMembers(Context context, ProgressBarCallBack progressBarCallBack, int queueId);

    LiveData<Queue> getQueue(int queueId);

    LiveData<Queue> getQueue(ProgressBarCallBack progressBarCallBack, String code);

    void createQueue(Context context, ProgressBarCallBack progressBarCallBack, CreateQueueModel createQueueModel);

    void joinQueue(Context context, ProgressBarCallBack progressBarCallBack, int queueId, JoinQueueRequestBody joinQueueRequestBody);

    void leaveQueue(Context context, ProgressBarCallBack progressBarCallBack, int queueId, int userId);

    void deleteQueue(Context context, ProgressBarCallBack progressBarCallBack, int queueId);
}

File: app/src/main/java/com/example/queue/repository/Repository.kt
package com.example.queue.repository

import com.example.queue.api.RetrofitInstance
import com.example.queue.models.CreateQueueRequestModel
import com.example.queue.models.LoginRequest
import com.example.queue.models.RegistrationRequest

class Repository {
    suspend fun login(loginRequest: LoginRequest) = RetrofitInstance.api.login(loginRequest)
    suspend fun registration(registrationRequest: RegistrationRequest) =
        RetrofitInstance.api.registration(registrationRequest)

    suspend fun logout(authorization: String) = RetrofitInstance.api.logout(authorization)
    suspend fun getLoggedInUser(authorization: String) =
        RetrofitInstance.api.getLoggedInUser(authorization)

    suspend fun getQueues() = RetrofitInstance.api.getQueues()
    suspend fun getQueue(code: String) = RetrofitInstance.api.getQueue(code)
    suspend fun createQueue(
        createQueueRequestModel: CreateQueueRequestModel,
        authorization: String
    ) = RetrofitInstance.api.createQueue(createQueueRequestModel, authorization)

    suspend fun getMembers(queueId: Int) = RetrofitInstance.api.getMembers(queueId)

}